Change PmData-SearchBar relation

SSZ4 opened the pull request:
SearchBar의 데이터를 PmData로 연동시키기 위해 몇 가지 방법을 시도 중입니다.

SSZ4 committed:
Change PmData-SearchBar relation

SearchBar의 데이터를 PmData로 연동시키기 위해 몇 가지 방법을 시도 중입니다.

luneclaire reviewed (changes requested):
검색한 주소 (SearchBar.js에서 addr)를 SearchBar.js -> PmData.js로 넘기면 나중에 검색창이 아니라 지도를 클릭하거나, 현재 위치를 가져올 시 해당 addr State를 다른 컴포넌트와 연결하기가 어려워질 것 같습니다.

App.js차원에서 setAddr등의 함수를 만들어서 이 함수를 PmData, SearchBar, 나중에는 map까지 하위 컴포넌트로 넘겨주는 형태로 addr을 관리하는 방법이 적합할 것 같아요.

다시 말해, addr에는 단순하게 검색창에 입력한 값만 들어가는 것이 아니라 클릭한 지도의 지역이나, 현재 위치 값도 넘겨줄 수 있게끔 관리하는 것이 좋을 것 같습니다. PmData에서는 검색을 통해 넘겨받거나, 지도를 클릭해 넘겨받은 addr에 대한 정보를 표시하는 것으로 하면 될 것 같습니다.

혹시 제 코멘트 중에 설명이 부족하거나 구현에 어려운 부분이 있으시면 말씀주세요!!

SSZ4 committed:
Make Switch Interaction

오늘/내일과 미세/초미세먼지 버튼과 상호작용 하도록 일시적으로 구성했습니다. 저번의 pull request에 관한 부분은 롤백했습니다. 버튼에 따라서 값을 받아오는 부분은 추후에 추가할 것입니다.

SSZ4 commented:
네 안그래도 그래서 그 부분 롤백 하고 app.js에서 중계하는 방향으로 다시 잡았습니다.

luneclaire reviewed (approved):
수정사항 확인했습니다! 수고하셨습니다 :)
